src/modules/image/imageApiInterfaces.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2020-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+interface Author {
+  name: string;
+  type: string;
+}
+
+interface Copyright {
+  license: {
+    license: string;
+    description: string;
+    url?: string;
+  };
+  origin: string;
+  processors: Author[];
+  rightsholders: Author[];
+  creators: Author[];
+  agreementId?: number;
+  validFrom?: string;
+  validTo?: string;
+}
```

Comment:
Blir ikkje dette duplisering? Eller meiner du at det er mest korrekt å separere api-interfaces.

Reply:
Forskjellige apier har forskjellige optional og required felter, det her blir mest clean løsning sånn sett. Jeg eksperimenterte også litt med bruk av `Omit<>` i listing-frontend. Prinsippet hadde vært å lage en "generisk" entity og extende og erstatte felter for å tilpasse til akkurat det man trenger, f.eks noe sånt:

```
import { Copyright } from '../../interfaces';

interface ImageCopyright extends Omit<Copyright, 'license'> {
  license: {
    license: string;
    description: string;
    url?: string;
  };
}
```

Her er `Copyright` en generisk entity som vi extender og erstatter `license` for å tilpasse det slik vi vil. `Copyright.license` for images skal kun ha url som optional, mens for artikkler f.eks er både url og description optional.

Vet egentlig hvilken løsning er best. Kom gjerne med forslag.

Reply:
Nah. Kan gjøre det sånn.